### PR TITLE
[Tooltip] disable empty tooltips

### LINKED ISSUE #886: Disable tooltips with empty content by default
There's no reason to show empty tooltips. I would prefer if Blueprint prevented me from shooting myself in the foot here:

image

If you agree @giladgray @llorca, I can send a PR

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- disable empty tooltips
- empty = null/undefined, empty string, whitespace strings.

### CODE REVIEW

File: packages/core/test/tooltip/tooltipTests.tsx [hunk 1]
```diff
@@ -52,7 +52,24 @@ describe("<Tooltip>", () => {
...
+                warnSpy.reset();
+            }
+
+            assertDisabledPopover("");
+            assertDisabledPopover("   ");
+            assertDisabledPopover(null);
+            warnSpy.restore();
+        });
+
         // it("setting isDisabled=true prevents opening tooltip", () => {
```
Comment (adidahiya) [inline packages/core/test/tooltip/tooltipTests.tsx]: unrelated: awkward commented out test

Comment (giladgray) [inline packages/core/test/tooltip/tooltipTests.tsx]: i know, and in trying to re-enable them i opened a whole can of worms.

**Comment by giladgray** [comment]
major blocker: this does not work for controlled `Tooltip`s because `isOpen` and `isDisabled` are mutually exclusive in `Popover`. I'm trying to unwind the mutex but it's proving very tricky.

**Comment by giladgray** [comment]
ok _now_ this PR should be acceptable. @adidahiya please re-review?

File: packages/core/test/popover/popoverTests.tsx [hunk 2]
```diff
@@ -551,6 +528,10 @@ describe("<Popover>", () => {
             </Popover>,
             { attachTo: testsContainerElement },
         ) as IPopoverWrapper;
+        wrapper.assertIsOpen = (isOpen = true) => {
+            assert.equal(wrapper.find(Overlay).prop("isOpen"), isOpen);
```
Comment (adidahiya) [inline packages/core/test/popover/popoverTests.tsx]: use `assert.strictEqual`. `assert.equal` should generally be banned

File: packages/core/src/components/popover/popover.tsx [hunk 3]
```diff
@@ -361,10 +361,6 @@ export class Popover extends AbstractComponent<IPopoverProps, IPopoverState> {
             console.warn(Errors.POPOVER_UNCONTROLLED_ONINTERACTION);
         }
 
-        if (props.isOpen != null && props.isDisabled) {
-            throw new Error(Errors.POPOVER_CONTROLLED_DISABLED);
```
Comment (adidahiya) [inline packages/core/src/components/popover/popover.tsx]: is this error string no longer used anywhere?

**Comment by adidahiya** [comment]
overall looks good though

File: packages/core/test/popover/popoverTests.tsx (hunk 2 above)
Comment (giladgray) [inline packages/core/test/popover/popoverTests.tsx]: agreed, but this one's a boolean

File: packages/core/src/components/popover/popover.tsx (hunk 3 above)
Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: indeed

Comment (adidahiya) [inline packages/core/src/components/popover/popover.tsx]: then let's remove it

Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: already done 👍

**Pull request merged**